// auth_service/grpc_pb/auth.proto

syntax = "proto3";

package custom_auth;

service AuthService {
    rpc Register(UserRegisterRequest) returns (UserRegisterResponse);
    rpc Login(UserLoginRequest) returns (UserLoginResponse);
    rpc RefreshToken(RefreshTokenRequest) returns (RefreshTokenResponse);
}

message UserRegisterRequest {
    string phone_number = 1;
    string password = 2;
}

message UserRegisterResponse {
    string user_id = 1;
    string message = 2;
}

message UserLoginRequest {
    string phone_number = 1;
    string password = 2;
}

message UserLoginResponse {
    string access_token = 1;
    string refresh_token = 2;
    string message = 3;
}

message RefreshTokenRequest {
    string refresh_token = 1;
}

message RefreshTokenResponse {
    string access_token = 1;
    string message = 2;
}
